### ISSUE TITLE
Warnings and errors analyzing Intel's NINDY 3.01 binary

### ISSUE THREAD

**Issue by biggestsonicfan** [mumbel/ghidra_i960#2]
Greetings again,
After realizing I had a binary of Intel i960 software dumped from some floppy disks I own, I decided to throw it into Ghidra 9.1.2 to see how this repo would fare against it. And it does pretty well, but I thought I would document warnings and errors along the way so they could be looked into at a future date:

**Warning:** Auto Analysis Summary: Function Start Search> Could not initialize a search state.
**Error:** Analyzer Error: Analysis Task: Stack - Bit length must be >= 1 and = 64

```
Analysis Task: Stack - Bit length must be >= 1 and <= 64
java.lang.IllegalArgumentException: Bit length must be >= 1 and <= 64
	at ghidra.program.model.scalar.Scalar.<init>(Scalar.java:62)
	at ghidra.program.util.VarnodeContext.extendValue(VarnodeContext.java:1267)
	at ghidra.program.util.SymbolicPropogator.applyPcode(SymbolicPropogator.java:1092)
	at ghidra.program.util.SymbolicPropogator.flowConstants(SymbolicPropogator.java:491)
	at ghidra.program.util.SymbolicPropogator.flowConstants(SymbolicPropogator.java:389)
	at ghidra.program.util.SymbolicPropogator.flowConstants(SymbolicPropogator.java:188)
	at ghidra.app.cmd.function.NewFunctionStackAnalysisCmd.createStackPointerVariables(NewFunctionStackAnalysisCmd.java:340)
	at ghidra.app.cmd.function.NewFunctionStackAnalysisCmd.analyzeFunction(NewFunctionStackAnalysisCmd.java:191)
	at ghidra.app.cmd.function.NewFunctionStackAnalysisCmd.applyTo(NewFunctionStackAnalysisCmd.java:118)
	at ghidra.app.plugin.core.function.StackVariableAnalyzer.added(StackVariableAnalyzer.java:54)
	at ghidra.app.plugin.core.analysis.AnalysisScheduler.runAnalyzer(AnalysisScheduler.java:185)
	at ghidra.app.plugin.core.analysis.AnalysisTask.applyTo(AnalysisTask.java:39)
	at ghidra.app.plugin.core.analysis.AutoAnalysisManager$AnalysisTaskWrapper.run(AutoAnalysisManager.java:685)
	at ghidra.app.plugin.core.analysis.AutoAnalysisManager.startAnalysis(AutoAnalysisManager.java:785)
	at ghidra.app.plugin.core.analysis.AutoAnalysisManager.startAnalysis(AutoAnalysisManager.java:664)
	at ghidra.app.plugin.core.analysis.AutoAnalysisManager.startAnalysis(AutoAnalysisManager.java:629)
	at ghidra.app.plugin.core.analysis.AnalysisBackgroundCommand.applyTo(AnalysisBackgroundCommand.java:62)
	at ghidra.framework.plugintool.mgr.BackgroundCommandTask.run(BackgroundCommandTask.java:101)
	at ghidra.framework.plugintool.mgr.ToolTaskManager.run(ToolTaskManager.java:315)
	at java.base/java.lang.Thread.run(Thread.java:832)

---------------------------------------------------
Build Date: 2020-Feb-12 1149 EST
Ghidra Version: 9.1.2
Java Home: /usr/lib64/jvm/java-15-openjdk-15
JVM Version: N/A 15-internal
OS: Linux 5.6.12-1-default amd64
Workstation: 0.0.0.0
```

The binary for NINDY 3.01 is uploaded as follows: [image.bin.zip](https://github.com/mumbel/ghidra_i960/files/4697264/image.bin.zip). Again, this is an Intel provided binary which should provide insight into improving repo as opposed to video game programs which may not be defined as ELF or COFF executables.

**Comment by mumbel** [comment]
Thanks for the sample.   I do remember coming across that as well.  I haven't a a clue what it is or what it means, guess I might need to run a debug build in eclipse.

**Comment by mumbel** [comment]
@cglmrfreeman  Maybe fixed.  2d297795a1e2f12698a34d555c351769f40b0fb1 

Not sure if this is a complete fix, but any `COBR` instruction format that used a immediate value had issues as it could not determine the number of bits of the value.

**Comment by mumbel** [comment]
@cglmrfreeman  came across your stack exchange post.  Looks like its hitting this same error when it decodes the scale (at least when i debug the instruction you have posted with DebugSleighInstructionParse.java).  I'll try to look into this a little more

maybe cb4513f8ae6d3009da53b10c70b536ddf520641b will help if youre still hitting `Bit length must be >= 1 and <= 64`

**Comment by biggestsonicfan** [comment]
Thanks much! Will check this out when I get a chance. I'm in the process of assembling an IDA disassembly and am discovering IDA omits quite a bit that maybe Ghidra could label properly. I'm sure once I am done with my disassembly producing an 1:1 binary with a newlib assembled disassembly, we could learn a lot from comparing Ghidra output to the disassembly.

**Comment by mumbel** [comment]
Sounds interesting, good luck

**Comment by biggestsonicfan** [comment]
Looks like the error I got is fixed when loading NINDY! I still get the warning, but I guess that's okay. Gonna close the issue now, I'll let you know if something new pops up!

**Comment by mumbel** [comment]
@cglmrfreeman  I think the warning was from 78225cf588d9538883972fb2486acd7689b81204, but I added a lot in 1a0e23927bcb7fc225c8b5c02d98230eec347763 since trying to repro that, so it might have been related to something there.  But can't get the warning anymore, auto-analysis seems to work with defaults.  Didn't find all the code, but did alot.

**Comment by mumbel** [comment]
@cglmrfreeman   sorry to ping again.  I added more commits today that should be helpful, at least for your particular binary from #1.  it should remove all cases of unimplemented instruction.  I made pcodeops for `scanbit` and `send`, its not doing anything with the instruction, but you'd see `__scanbit` or `__send` in decompiler at least.  Absolute addressing is now unsigned (etc for accessing reserved high 12MB, you'll see 0xff000010 instead of -0xfffff70 or whatever it'd be), and `bal` and `balx` should have better flow control during disassembly/auto-analysis.
